packages/cli/src/commands/rewards/slasher.ts

```diff
@@ -0,0 +1,108 @@
+import { Address } from '@celo/contractkit/lib/base'
+import { ValidatorsWrapper } from '@celo/contractkit/lib/wrappers/Validators'
+import { bitIsSet, parseBlockExtraData } from '@celo/utils/lib/istanbul'
+import { BaseCommand } from '../../base'
+
+export default class Slasher extends BaseCommand {
+  static description = 'Mines for slashable downtime'
+
+  static flags = {
+    ...BaseCommand.flags,
+  }
+
+  static args = []
+
+  static examples = ['slasher']
+
+  async run() {
+    // const res = this.parse(Slasher)
+    const validators = await this.kit.contracts.getValidators()
+    const downtimeSlasher = await this.kit.contracts.getDowntimeSlasher()
+    const slashableDowntime = await downtimeSlasher.slashableDowntime()
+    let validatorDownSince: number[] = []
+
+    let blockNumber = (await this.kit.web3.eth.getBlockNumber()) - 1
+    let epochNumber = -2
+    while (true) {
```

Comment:
Probably because i've been reviewing this for a while now but i'm having a hard time following this piece of code. In any case, i would recommend having small function to do the different tasks, each of them with name that cleary states the intent :D